edx_proctoring/views.py

```diff
@@ -956,6 +957,94 @@ def post(self, request):
+
+    Supports:
+        HTTP GET: return information about learners' attempts
+
+    **Expected Response**
+        HTTP GET:
+            The response will contain a dictionary with pagination info and a key `proctored_exam_attempts`
+            * proctored_exam_attempts: a list of dictionaries, where each dictionary contains all fields
+              for the most current exam attempt, and a key `all_attempts`, whose value
+              is a list of all attempts associated with a user and exam
+
+    **Exceptions**
+        HTTP GET:
+            * 403 if the requesting user is not staff or course staff for the course associated with
+            the supplied course ID
+    """
+    @method_decorator(require_course_or_global_staff)
+    def get(self, request, course_id, search_by=None):
+        """
+        HTTP GET Handler.
+        """
+        if search_by is not None:
+            exam_attempts = ProctoredExamStudentAttempt.objects.get_filtered_exam_attempts(
+                course_id, search_by
+            )
+            attempt_url = reverse('edx_proctoring:proctored_exam.attempts.grouped.search', args=[course_id, search_by])
+        else:
+            exam_attempts = ProctoredExamStudentAttempt.objects.get_all_exam_attempts(
+                course_id
+            )
+            attempt_url = reverse('edx_proctoring:proctored_exam.attempts.grouped.course', args=[course_id])
+
+        # get the most recent attempt for each unique user/exam combination
+        most_recent_attempts = list(self._get_newest_created_exam_attempt_per_user(exam_attempts).values())
+
+        paginator = Paginator(most_recent_attempts, ATTEMPTS_PER_PAGE)
+        page = request.GET.get('page')
+        exam_attempts_page = paginator.get_page(page)
+
+        grouped_attempts = []
+
+        for attempt in exam_attempts_page.object_list:
+            # serialize data
+            serialized_attempt = ProctoredExamStudentAttemptSerializer(attempt).data
+            user_id = serialized_attempt['user']['id']
+            exam_id = serialized_attempt['proctored_exam']['id']
+            # add all attempts that aren't the most recently created
+            grouped_past_attempts = get_user_attempts_by_exam_id(user_id, exam_id)
+            serialized_attempt['all_attempts'] = grouped_past_attempts
+            grouped_attempts.append(serialized_attempt)
+
+        response_data = {
+            'proctored_exam_attempts': grouped_attempts,
+            'pagination_info': {
+                'has_previous': exam_attempts_page.has_previous(),
+                'has_next': exam_attempts_page.has_next(),
+                'current_page': exam_attempts_page.number,
+                'total_pages': exam_attempts_page.paginator.num_pages,
+            },
+            'attempt_url': attempt_url
+
+        }
+        return Response(response_data)
+
+    def _get_newest_created_exam_attempt_per_user(self, attempts):
```

Comment:
NIT: thinking we should just rename this to get_first_exam_... or something. "Newest created" is misleading if you're reading code that calls this function.